src/vma/dev/allocator.cpp

```diff
@@ -93,30 +83,32 @@ void* vma_allocator::alloc_and_reg_mr(size_t size, ib_ctx_handler *p_ib_ctx_h)
 		else {
 			__log_info_dbg("Huge pages allocation passed successfully");
 			if (g_p_ib_ctx_handler_collection->get_ib_cxt_list() &&
-					!register_memory(size, p_ib_ctx_h, m_non_contig_access_mr)) {
+					!register_memory(size, p_ib_ctx_h, access)) {
 				__log_info_dbg("failed registering huge pages data memory block");
 				throw_vma_exception("failed registering huge pages data memory"
 						" block");
 			}
+			m_mem_alloc_type = ALLOC_TYPE_HUGEPAGES;
 			break;
 		}
 	// fallthrough
 	case ALLOC_TYPE_CONTIG:
-		if (m_is_contig_alloc &&
-				g_p_ib_ctx_handler_collection->get_ib_cxt_list()) {
-			if (!register_memory(size, p_ib_ctx_h, m_contig_access_mr)) {
+#ifdef VMA_IBV_ACCESS_ALLOCATE_MR
+		if (!safe_mce_sys().cpuid_hv() && g_p_ib_ctx_handler_collection->get_ib_cxt_list()) {
```

Comment:
Why not use safe_mce_sys().is_hypervisor ?

Reply:
no reason. will update